README.md

```diff
@@ -1,2 +1,100 @@
 # Rest7Contract
-Repository to discuss the new REST API contract for DSpace 7 https://wiki.duraspace.org/display/DSPACE/DSpace+7+UI+Working+Group
+Repository to discuss the new REST API contract for DSpace 7 <https://wiki.duraspace.org/display/DSPACE/DSpace+7+UI+Working+Group>
+
+## Use of the HTTP Verbs and HTTP Response CODE
+
+### On collection of resources endpoints
+- POST
+Adds a new element to the collection.
+
+- GET
+Returns the first page of the resources in the collection
+
+### On single resource endpoints
+- GET
+Returns a single entity.
+
+- HEAD
+Returns whether the item resource is available.
+
+- PUT
+Replaces the state of the target resource with the supplied request body.
+
+- PATCH
+Similar to PUT but partially updating the resources state.
+
+- DELETE
+Deletes the resource exposed.
+
+### On sub-path of a single resource endpoint)
+- GET
+Returns the state of the association resource
+
+- PUT
+Binds the resource pointed to by the given URI(s) to the resource. Return 400 Bad Request if multiple URIs were given for a to-one-association
+
+- POST
+Only supported for collection associations. Adds a new element to the collection.
+
+- DELETE
+Unbinds the association. Return 405 Method Not Allowed if the association is non-optional
+
+### Error codes
+400 Bad Request - if multiple URIs were given for a to-one-association
+
+404 Not found - if the requested entity or collection doesn't exists
+
+405 Method Not Allowed - if the methods is not implemented or a DELETE methods is called on a non-optional association
+
+## HATEOAS & HAL
+The new REST DSpace API supports the HATEOAS paradigm and adopt the HAL format to express links and embedded resources.
+
+## Pagination
+Each endpoints that expose a collection of resources, including sub-paths for embedded or linked collections (aka list of items of a collection, etc.), MUST implement the pagination with the following common behavior
+
+### Request parameter
+- **page**: 0 based integer value that specify the requested page in the result set [default 0]
+- **size**: the dimension of the result set window to show. It must be a positive value. Negative or zero value must be rejected with a 400 Error code. The default value as well as maximum values are configurable by the system administrator. Different Maximum values apply for anonymous users, logged-in users and administrators. Size over the maximum value are automatically reset to the maximum allowed value, no error is thrown.
+- **sort**: the criteria to use. Ordering is specified appending a comma and the keyword asc or desc to the criteria name (i.e. title,asc). Unknown sort criteria and/or ordering keyword produce an error response with Http Code 400
+
+### Response
+The HAL document always includes the following attributes
```

Comment:
HAL also returns a page object that contains the following properties: size, totalElements, totalPages, number.